tests/e2e/system_test/test_graceful_nodes_shutdown.py

```diff
@@ -0,0 +1,713 @@
+"""
+Test to verify that data is accessible and uncorrupted as well as
+operational cluster after graceful nodes shutdown
+"""
+import re
+import boto3
+import json
+import logging
+import pytest
+import time
+import uuid
+
+from datetime import datetime
+from semantic_version import Version
+
+from ocs_ci.ocs.ocp import OCP, wait_for_cluster_connectivity
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources import pod
+from ocs_ci.ocs.resources.rgw import RGW
+from ocs_ci.ocs.resources.objectbucket import OBC
+from ocs_ci.ocs.resources.pod import get_pod_logs, get_rgw_pods, get_pod_obj
+from ocs_ci.framework.testlib import E2ETest
+from ocs_ci.framework import config
+from ocs_ci.framework.pytest_customization.marks import (
+    system_test,
+    polarion_id,
+    skipif_no_kms,
+    skipif_ocs_version,
+)
+from ocs_ci.helpers.sanity_helpers import Sanity
+from ocs_ci.ocs.node import (
+    get_nodes,
+)
+from ocs_ci.utility.utils import exec_cmd, run_cmd, clone_notify
+from ocs_ci.ocs.utils import get_pod_name_by_pattern
+from ocs_ci.helpers.helpers import default_storage_class
+from ocs_ci.ocs.bucket_utils import s3_put_object, retrieve_verification_mode
+
+from ocs_ci.ocs.resources.fips import check_fips_enabled
+from ocs_ci.ocs.jenkins import Jenkins
+from ocs_ci.ocs.cosbench import Cosbench
+from ocs_ci.ocs.quay_operator import QuayOperator
+from ocs_ci.ocs.amq import AMQ
+from ocs_ci.ocs.constants import AWSCLI_TEST_OBJ_DIR
+from ocs_ci.ocs.bucket_utils import (
+    compare_bucket_object_list,
+    sync_object_directory,
+    write_random_test_objects_to_bucket,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class TestGracefulNodesShutdown(E2ETest):
+    """
+    Test uncorrupted data and operational cluster after graceful nodes shutdown
+    """
+
+    amq = None
+    cosbench = None
+    jenkins = None
+    quay_operator = None
+    bucket_names_list = []
+    mcg_obj = None
+    mcg_obj_session = None
+
+    @pytest.fixture(autouse=True)
+    def checks(self):
+        # This test is skipped due to https://issues.redhat.com/browse/ENTMQST-3422
+        check_fips_enabled()
+
+        nodes = get_nodes()
+        for node in nodes:
+            assert (
+                node.get()["status"]["allocatable"]["hugepages-2Mi"] == "64Mi"
+            ), f"Huge pages is not applied on {node.name}"
+
+    @pytest.fixture(autouse=True)
+    def setup(
+        self,
+        request,
+        project_factory,
+        pv_encryption_kms_setup_factory,
+        storageclass_factory,
+        pvc_factory,
+        pod_factory,
+        pvc_clone_factory,
+        snapshot_factory,
+        snapshot_restore_factory,
+        mcg_obj,
+        bucket_factory,
+        install_logging,
```

Comment:
Since OCP 4.12 is not yet GAed we will have to skip running the logging test on 4.12.This test will fail on 4.12, please check